Toolbox/arcsdm/areafrequency.py

```diff
@@ -39,369 +38,257 @@
-            tblrows.UpdateRow(tblrow)
+            tblrows.updateRow(tblrow)
             effarea.append(area)
             nSites.append(frequency)
             stats_found += 1
-            # gp.AddMessage("Debug: Stats_found =  %s"%stats_found);
         
-        tblrow = tblrows.Next()
+        tblrow = tblrows.next()
     
-    del tblrow,tblrows
-    gp.AddMessage('stats_found: %s'%stats_found)
-    # Check that output table is consistent with statistics
+    del tblrow, tblrows
+    arcpy.AddMessage('stats_found: %s' % stats_found)
     if stats_found < len(stats_dict):
-        gp.adderror('Not enough Values with Frequency > 0 found!')
+        arcpy.AddError('Not enough Values with Frequency > 0 found!')
         assert False
     elif stats_found > len(stats_dict):
-        gp.adderror('Too many Values with Frequency > 0 found!')
+        arcpy.AddError('Too many Values with Frequency > 0 found!')
         assert False
-    else:
-        pass
-        # gp.addmessage('All Values with Frequency > 0 found')
-
-    # From two reversed lists, create two lists and two cumulative summations
-    # gp.AddMessage("Calculating CAPP_CumAre,Eff_CumAre,Cum_Sites,I_CumSites fields...")
-    effarea_rev = reversed(effarea) # generator
-    nSites_rev = reversed(nSites) # generator
-    effCumarea=0
-    cumSites=0
+    
+    # Calculate cumulative areas and sites
+    effarea_rev = reversed(effarea)
+    nSites_rev = reversed(nSites)
+    effCumarea = 0
+    cumSites = 0
     effCumareaList = []
-    cumSitesList = [] 
+    cumSitesList = []
     for i in range(len(nSites)):
         effCumarea += 100.0 * next(effarea_rev) / totalarea
         effCumareaList.append(effCumarea)
         cumSites += 100.0 * next(nSites_rev) / totalsites
         cumSitesList.append(cumSites)
-        
-    # Update four fields from reversed lists
-    effCumareaList_rev = reversed(effCumareaList) # generator
-    cumSitesList_rev = reversed(cumSitesList) # generator 
-    # gp.AddMessage('doing update....')
-    tblrows = gp.UpdateCursor(Output_Table)
-    # gp.AddMessage(str(tblrows))
-    tblrow = tblrows.Next()
+    
+    effCumareaList_rev = reversed(effCumareaList)
+    cumSitesList_rev = reversed(cumSitesList)
+    
+    # Update the output table with cumulative values
+    tblrows = arcpy.UpdateCursor(Output_Table)
+    tblrow = tblrows.next()
     while tblrow:
-        # gp.AddMessage(str(tblrow) + str(i))
         cumArea += 100.0 * tblrow.Area_sqkm / totalarea
         tblrow.CAPP_CumAr = cumArea
         tblrow.Eff_CumAre = next(effCumareaList_rev)
         Cum_Sites = next(cumSitesList_rev)
         tblrow.Cum_Sites = Cum_Sites
-        tblrow.SetValue('I_CumSites', 100.0 - Cum_Sites)
-        tblrows.UpdateRow(tblrow)
-        tblrow = tblrows.Next()
-    # gp.addmessage('done.')
+        tblrow.setValue('I_CumSites', 100.0 - Cum_Sites)
+        tblrows.updateRow(tblrow)
+        tblrow = tblrows.next()
+    
     del tblrow, tblrows
```

Comment:
I would replace this with:
```
    with arcpy.da.UpdateCursor(Output_Table, ["RASTERVALU", "Area_sqkm", "CAPP_CumAr", "Eff_CumAre", "Cum_Sites", "I_CumSites"]) as cursor:
        for row in cursor:
            cumArea += 100.0 * row[1] / totalarea
            row[2] = cumArea
            row[3] = next(effCumareaList_rev)
            Cum_Sites = next(cumSitesList_rev)
            row[4] = Cum_Sites
            row[5] = 100.0 - Cum_Sites
            cursor.updateRow(row)
```

Reply:
I have edited the code and commited new code.